.travis.yml

```diff
@@ -50,7 +50,7 @@ jobs:
       before_install:
         - pip install pre-commit==2.5.1
       script:
-        - pre-commit run --origin HEAD --source origin/HEAD --show-diff-on-failure
+        - pre-commit run --from-ref HEAD^1 --to-ref HEAD --show-diff-on-failure
```

Comment:
Won't this just compare the current with the previous commit?

From https://pre-commit.com/#usage-in-continuous-integration:
> To check only files which have changed, which may be faster, use something like `pre-commit run --from-ref origin.HEAD --to-ref HEAD`.

Reply:
Travis is checking out the PR commit. That is a virtual merge commit that merges the PR branch with the merge base. The PR Branch is HEAD^2 and the merge base at HEAD^1. 
pre-commit uses diff HEAD^1...HEAD, which crawls automatically up. 
The only "issue" is that Travis did not retest the actual merge, before pressing the merge button if the HEAD^1 as advanced. I will add a comment.

Reply:
origin/HEAD is the head of the default branch (master in our case). That's why we have false positives.